luthier/src/main/java/com/yahoo/bard/webservice/config/luthier/factories/DefaultMetricFactory.java

```diff
@@ -0,0 +1,78 @@
+import com.yahoo.bard.webservice.config.luthier.LuthierValidationUtils;
+import com.yahoo.bard.webservice.data.config.metric.makers.MetricMaker;
+import com.yahoo.bard.webservice.data.metric.LogicalMetric;
+import com.yahoo.bard.webservice.data.metric.LogicalMetricInfo;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * A Factory that supports all aggregation metrics, including but not limited to: longSum, doubleMax, longMin, etc.
+ */
+public class DefaultMetricFactory implements Factory<LogicalMetric> {
+    private static final String ENTITY_TYPE = "AggregationMetric";
+    private static final String MAKER = "maker";
+    private static final String LONG_NAME = "longName";
+    private static final String CATEGORY = "category";
+    private static final String DESCRIPTION = "description";
+    private static final String DATA_TYPE = "dataType";
+    private static final String DEPENDENCY_METRIC_NAMES = "dependencyMetricNames";
+
+    @Override
+    public LogicalMetric build(String name, ObjectNode configTable, LuthierIndustrialPark resourceFactories) {
+        validateFields(
+                name,
+                configTable,
+                Arrays.asList(MAKER, LONG_NAME, CATEGORY, DESCRIPTION, DEPENDENCY_METRIC_NAMES)
+        );
+        MetricMaker maker = resourceFactories.getMetricMaker(configTable.get(MAKER).textValue());
+        LogicalMetricInfo metricInfo = configTable.get(DATA_TYPE) == null ?
+                new LogicalMetricInfo(
+                        name,
+                        configTable.get(LONG_NAME).textValue(),
+                        configTable.get(CATEGORY).textValue(),
+                        configTable.get(DESCRIPTION).textValue()
+                )
+                : new LogicalMetricInfo(
+                        name,
+                        configTable.get(LONG_NAME).textValue(),
+                        configTable.get(CATEGORY).textValue(),
+                        configTable.get(DESCRIPTION).textValue(),
+                        configTable.get(DATA_TYPE).textValue()
+                );
+        List<String> dependencyMetricNames = StreamSupport.stream(
+                configTable.get(DEPENDENCY_METRIC_NAMES).spliterator(),
+                false
+        )
+                .map(JsonNode::textValue)
+                .collect(Collectors.toList());
+        return maker.make(metricInfo, dependencyMetricNames);
+    }
+    /**
+     * Helper function to validate only the fields needed in the parameter build.
+     *
+     * @param name  the config dictionary name (normally the apiName)
+     * @param configTable  ObjectNode that points to the value of corresponding table entry in config file
+     * @param fieldNames  the list of field names we want to validate existence in this configTable
+     */
+    private void validateFields(String name, ObjectNode configTable, List<String> fieldNames) {
```

Comment:
If you wanted, you could have this method take a variable number of arguments: `validateFields(String name, ObjectNode configTable, String fieldNames...)`
Then you could call it with something like `validateFields(name, config, FIELD_NAME_1, FIELD_NAME_2, FIELD_NAME_3)` without having to wrap the field names
in a list.

Not a blocker.

Reply:
This method is lifted to the LuthierValidationUtils due to about 6 Factories are using it. As a result the code duplication massively reduced.